tpu-client-next/src/connection_workers_scheduler.rs

```diff
@@ -114,50 +133,49 @@ impl ConnectionWorkersScheduler {
                     break;
                 }
             };
-            let updated_leaders = leader_updater.next_leaders(lookahead_slots);
-            let new_leader = &updated_leaders[0];
-            let future_leaders = &updated_leaders[1..];
-            if !workers.contains(new_leader) {
-                debug!("No existing workers for {new_leader:?}, starting a new one.");
-                let worker = Self::spawn_worker(
-                    &endpoint,
-                    new_leader,
-                    worker_channel_size,
-                    skip_check_transaction_age,
-                    max_reconnect_attempts,
-                );
-                workers.push(*new_leader, worker).await;
-            }
 
-            tokio::select! {
-                send_res = workers.send_transactions_to_address(new_leader, transaction_batch) => match send_res {
-                    Ok(()) => (),
-                    Err(WorkersCacheError::ShutdownError) => {
-                        debug!("Connection to {new_leader} was closed, worker cache shutdown");
-                    }
-                    Err(err) => {
-                        warn!("Connection to {new_leader} was closed, worker error: {err}");
-                       // If we has failed to send batch, it will be dropped.
-                    }
-                },
-                () = cancel.cancelled() => {
-                    debug!("Cancelled: Shutting down");
-                    break;
-                }
-            };
+            let updated_leaders = leader_updater.next_leaders(leaders_fanout.connect);
 
-            // Regardless of who is leader, add future leaders to the cache to
-            // hide the latency of opening the connection.
-            for peer in future_leaders {
+            let (fanout_leaders, connect_leaders) =
+                split_leaders(&updated_leaders, &leaders_fanout);
```

Comment:
It is a bit confusing that we call it `send` and `connect` portions elsewhere, but here they are called `fanout` and `connection` portions.

Maybe it would be more consistent to call it `send_leaders` and `connect_leaders` here as well?

```suggestion
            let (send_leaders, connect_leaders) =
                split_leaders(&updated_leaders, &leaders_fanout);
```